msgpush/src/main/java/com/gcinterceptor/msgpush/MsgPush.java

```diff
@@ -0,0 +1,64 @@
+package com.gcinterceptor.msgpush;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+import com.gcinterceptor.spring.SpringGcInterceptor;
+import com.gcinterceptor.spring.SpringGcInterceptorConfiguration;
+
+@SpringBootApplication
+@Import(SpringGcInterceptorConfiguration.class)
+public class MsgPush {
+	private static final boolean USE_GCI = Boolean.parseBoolean(System.getenv("use_gci"));
+	private static int MSG_SIZE = Integer.parseInt(System.getenv("msg_size"));
+	private static int WINDOWS_SIZE = Integer.parseInt(System.getenv("windows_size"));
+	private static byte[][] buffer;
+	private static int msgCount;
+
+	public static void main(String[] args) {
+		SpringApplication.run(MsgPush.class, args);
+	}
+
+	@Bean
+	public WebMvcConfigurer adapter() {
+		return new WebMvcConfigurer() {
+			@Autowired
+			SpringGcInterceptor gci;
+			
+			@Override
+			public void addInterceptors(InterceptorRegistry registry) {
+				if (USE_GCI) {
+					registry.addInterceptor(gci);
+				}
+			}
+		};
+	}
+
+	@RestController
+	public static class HelloController {
+		@RequestMapping("/")
+		public void index() throws InterruptedException {
+			buffer = new byte[WINDOWS_SIZE][MSG_SIZE];
```

Comment:
You should not create a new buffer at every iteration. This should be created in a static block. Along with you populate the variables.

```java
private static int MSG_SIZE;
private static int WINDOWS_SIZE;
private static byte[][] buffer;
static {
    WINDOW_SIZE = Integer.parseInt(System.getenv("window_size"));
    MSG_SIZE = Integer.parseInt(System.getenv("msg_size"));
    buffer = new byte[WINDOWS_SIZE][MSG_SIZE];
}
```